subsys/net/lib/zstream/zstream_tls.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <sys/types.h>
+#include <stdbool.h>
+#include <errno.h>
+
+#if !defined(CONFIG_MBEDTLS_CFG_FILE)
+#include "mbedtls/config.h"
+#else
+#include CONFIG_MBEDTLS_CFG_FILE
+#endif /* CONFIG_MBEDTLS_CFG_FILE */
+#include "mbedtls/platform.h"
+
+#include <mbedtls/net.h>
```

Comment:
This header appears to be deprecated, and superseded by mbedtls/net_sockets.h.

It is included in a few places in this PR:

```
In file included from /home/mbolivar/src/zephyr/include/net/zstream_tls.h:23:0,
                 from src/big_http_download.c:32:
/usr/include/mbedtls/net.h:29:2: warning: #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h" [-Wcpp]
 #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h"
  ^~~~~~~
In file included from /home/mbolivar/src/zephyr/subsys/net/lib/zstream/zstream_tls.c:17:0:
/usr/include/mbedtls/net.h:29:2: warning: #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h" [-Wcpp]
 #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h"
  ^~~~~~~
In file included from /home/mbolivar/src/zephyr/include/net/zstream_tls.h:23:0,
                 from /home/mbolivar/src/zephyr/subsys/net/lib/zstream/zstream_tls.c:34:
/usr/include/mbedtls/net.h:29:2: warning: #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h" [-Wcpp]
 #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h"
  ^~~~~~~
In file included from /home/mbolivar/src/zephyr/subsys/net/lib/zstream/../tls_conf/tls_conf.c:17:0:
/usr/include/mbedtls/net.h:29:2: warning: #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h" [-Wcpp]
 #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h"
  ^~~~~~~
```

I got the warnings when building big_http_download on a POSIX host.

The version of this file in Zephyr's bundled mbedTLS itself just includes net_sockets.h, so it seems safe to fix the deprecation warnings on hosts where they are generated.

Reply:
I'm using well-deployed and supported Ubuntu 16.04LTS, it ships mbedTLS 2.2.1 and lacks mbedtls/net_sockets.h . So, for the purpose of compatibility with different POSIX hosts, I'd be reluctant to make such a change.

However, a quick try to remove that include seems to work, for both Ubuntu's 2.2.1 and Zephyr's 2.8.0. That's a bit strange, because what's needed from that header are error defines from MBEDTLS_ERR_NET_SEND_FAILED, and quickly looking again, I don't see what else pulls them, but apparently something does. So, let me just try to remove it.

Reply:
Forgot to mention that how it got there in the first place was copy-paste from net_app ;-).

Reply:
I see. I tested on Ubuntu 17.10, which ships with mbedtls 2.5.1. I'm fine either way, just thought I would point it out.